### ISSUE TITLE
AI players should prefer deep water colonies, at least if they have none already

### ISSUE THREAD

**Issue by Nightinggale** [We-the-People-civ4col-mod/Mod#933]
From the chat
> The Portuguese here has settled on shallow coast, jeopardizing all their colonization effort, ended up only have 1 city ever since then.

So AI ended up with a shallow water harbor, couldn't figure out what to do with the deep water ships and got stuck.

![image](https://github.com/We-the-People-civ4col-mod/Mod/assets/4641647/137c7c57-d3b3-4e61-b220-a92ad30f594a)

**Comment by raystuttgart** [comment]
Actually there **already** is logic that tells it to prefer settling on deep water coast. :)
(I wrote it, when I implemented the feature.)

It is just that sometimes it still for some reason prefers to settle o other places ...
But theoretically it would be also possible to 100% forbid it to settle its first City on Plots that have no "Deep Water Coast".

**Comment by Nightinggale** [comment]
Alternatively the AI should learn how to handle ships, which can't reach the colony.

**Comment by raystuttgart** [comment]
This is the logic I had written.
I wrote it for Large Rivers, but it works for Shallow Coast as well.

```cpp
if (pPlot->isCoastalLand(GC.getMIN_WATER_SIZE_FOR_OCEAN()))
	{
		//WTP, ray, Large Rivers - START
		// to ensure that first city is not found at Large River without other Water access
		if (bStartingLoc)
		{
			if (pPlot->hasDeepWaterCoast())
			{
				iValue *= 125; // found value increased
				iValue /= 100;
			}
			else
			{
				iValue *= 80; // found value decreased
				iValue /= 100;
			}
		}

		// old normal logic - not considering special cases for Large Rivers without direct Ocean Plots - for all Cities after first city
		else
		{
			iValue *= 125;
			iValue /= 100;
		}
		//WTP, ray, Large Rivers - END
	}
```

But maybe somebody wants to further optimize it. 👍

**Comment by raystuttgart** [comment]
Maybe it will even be enough to change these here:
(The logic I posted above is just for the **Starting City**.)

```
iValue *= 125; // found value increased
....
iValue *= 80; // found value decreased
```

Values like this might be better:

```
iValue *= 200; // found value increased
....
iValue *= 50; // found value decreased
```

**Comment by raystuttgart** [comment]
> Alternatively the AI should learn how to handle ships, which can't reach the colony.

Normally it can ... but it is possible that e.g. its Caravel gets killed and it only has a Brigantine (or whatever) left ...
Otherwise the problem mentioned above would happen much more often. :)

**Comment by Nightinggale** [comment]
Maybe the AI should prefer shallow water ships if said player only has shallow water harbors.

**Comment by raystuttgart** [comment]
> Maybe the AI should prefer shallow water ships if said player only has shallow water harbors.

Yeah, that could be an AI improvement as well, but my last change might also already make things better.
(As AI will have a much stronger tendency to settle **first City** at DeepWaterCoast.)

**Comment by devolution79** [comment]
https://github.com/We-the-People-civ4col-mod/Mod/tree/generatePathForHypotheticalUnit  is essential for addressing this issue, as well as others of a similar nature. It's crucial because AI settlers need to carry out pathfinding at the very least to ascertain the viability of a location for city placement.

It's worth pointing out that the various adjacent plot checks that we currently have are in my opinion,  completely inadequate for this purpose. We really need full pathfinding in these circumstance and we need to be able to do so without an actual unit being present.

**Comment by Nightinggale** [comment]
861ed80889e4ac28496fb1fda85eb2d6dfee7756 seems to have fixed this. If an AI has no colonies, it can only found on plots next to deep water.
